extensions/distinctcount/src/main/java/io/druid/query/aggregation/distinctcount/DistinctCountAggregatorFactory.java

```diff
@@ -0,0 +1,214 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.distinctcount;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.primitives.Longs;
+import com.metamx.common.StringUtils;
+import com.metamx.common.logger.Logger;
+import io.druid.query.aggregation.Aggregator;
+import io.druid.query.aggregation.AggregatorFactory;
+import io.druid.query.aggregation.BufferAggregator;
+import io.druid.query.aggregation.LongSumAggregatorFactory;
+import io.druid.query.dimension.DefaultDimensionSpec;
+import io.druid.segment.ColumnSelectorFactory;
+import io.druid.segment.DimensionSelector;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+public class DistinctCountAggregatorFactory implements AggregatorFactory
+{
+  private static final Logger log = new Logger(DistinctCountAggregatorFactory.class);
+  private static final byte CACHE_TYPE_ID = 20;
+  private static final BitMapFactory DEFAULT_BITMAP_FACTORY = new RoaringBitMapFactory();
+
+  private final String name;
+  private final String fieldName;
+  private final BitMapFactory bitMapFactory;
+
+  @JsonCreator
+  public DistinctCountAggregatorFactory(
+      @JsonProperty("name") String name,
+      @JsonProperty("fieldName") String fieldName,
+      @JsonProperty("bitmap") BitMapFactory bitMapFactory
+  )
+  {
+    this.name = name;
+    this.fieldName = fieldName;
+    this.bitMapFactory = bitMapFactory == null ? DEFAULT_BITMAP_FACTORY : bitMapFactory;
+  }
+
+  @Override
+  public Aggregator factorize(ColumnSelectorFactory columnFactory)
+  {
+    return new DistinctCountAggregator(
+        name,
+        makeDimensionSelector(columnFactory),
+        bitMapFactory.makeEmptyMutableBitmap()
+    );
+  }
+
+  @Override
+  public BufferAggregator factorizeBuffered(ColumnSelectorFactory columnFactory)
+  {
+    return new DistinctCountBufferAggregator(makeDimensionSelector(columnFactory));
+  }
+
+  private DimensionSelector makeDimensionSelector(final ColumnSelectorFactory columnFactory)
+  {
+    return columnFactory.makeDimensionSelector(new DefaultDimensionSpec(fieldName, fieldName));
+  }
+
+  @Override
+  public Comparator getComparator()
+  {
+    return new Comparator()
+    {
+      @Override
+      public int compare(Object o, Object o1)
+      {
+        return Longs.compare(((Number) o).longValue(), ((Number) o1).longValue());
+      }
+    };
+  }
+
+  @Override
+  public Object combine(Object lhs, Object rhs)
+  {
+    return ((Number) lhs).longValue() + ((Number) rhs).longValue();
```

Comment:
Can it be possible to merge two distinct count just by adding them? Am I missing something?